services/DevHome.Services.DesiredStateConfiguration/Services/DSCOperations.cs

```diff
@@ -44,8 +45,32 @@ public async Task<IDSCSet> GetConfigurationUnitDetailsAsync(IDSCFile file)
         var configSet = await OpenConfigurationSetAsync(file, processor);
 
         _logger.LogInformation("Getting configuration unit details");
-        await processor.GetSetDetailsAsync(configSet, ConfigurationUnitDetailFlags.ReadOnly);
-        return new DSCSet(configSet);
+        var detailsOperation = processor.GetSetDetailsAsync(configSet, ConfigurationUnitDetailFlags.ReadOnly);
+        var detailsOperationTask = detailsOperation.AsTask();
+
+        var set = new DSCSet(configSet);
+
+        // For each DSC unit, create a task to get the details asynchronously
+        // in the background
+        foreach (var unit in set.UnitsInternal)

```

Comment:
Why not use a Parallel.ForEach loop here?

Reply:
Technically I could set those in parallel, but I don't suspect performance improvement here because this is pretty much an assignment expression where the right-hand side is a `Task` value that is not being awaited, for a small number of units.